dotCMS/src/main/java/com/dotcms/vanity/DefaultVanityUrlHandler.java

```diff
@@ -0,0 +1,92 @@
+package com.dotcms.vanity;
+
+import com.dotcms.content.model.VanityUrl;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.db.HibernateUtil;
+import com.dotmarketing.filters.CMSFilter;
+import com.dotmarketing.filters.CmsUrlUtil;
+import com.dotmarketing.util.InodeUtils;
+import com.dotmarketing.util.UtilMethods;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * This class implements the methods defined in the {@link VanityUrlHandler}
+ * Created by oswaldogallango on 2017-06-16.
+ */
+public class DefaultVanityUrlHandler implements VanityUrlHandler{
+
+    CmsUrlUtil urlUtil = CmsUrlUtil.getInstance();
+
+    @Override
+    public VanityUrlResult handle(VanityUrl vanityUrl, HttpServletResponse response, Host host, long languageId) throws IOException{
+        String rewrite = null;
+        String queryString = null;
+        CMSFilter.IAm iAm = null;
+
+        if(vanityUrl != null){
+            rewrite = InodeUtils.isSet(vanityUrl.getInode())?vanityUrl.getForwardTo():null;
+
+            if(vanityUrl.getAction() == HttpServletResponse.SC_OK){
+                //then forward
+                response.setStatus(vanityUrl.getAction());
+            }else if(vanityUrl.getAction() == HttpServletResponse.SC_MOVED_PERMANENTLY || vanityUrl.getAction() == HttpServletResponse.SC_FOUND){
+                //redirect
+                response.setStatus(vanityUrl.getAction());
+                response.sendRedirect(rewrite);
+
+                closeDbSilently();
+                return new VanityUrlResult(null,null,null,true);
```

Comment:
This could be a constants such as return VanityUrlResult.DEFAULT_RESULT

and

private final static VanityUrlResult  DEFAULT_RESULT = new VanityUrlResult(null,null,null,true);

It could safe some memory since it is gonna be use a lot